rust/operator-binary/src/csi_server/controller.rs

```diff
@@ -0,0 +1,269 @@
+use std::collections::BTreeMap;
+
+use crate::{
+    backend::{self, SecretBackendError, SecretVolumeSelector},
+    grpc::csi::{
+        self,
+        v1::{
+            controller_server::Controller, controller_service_capability,
+            ControllerGetCapabilitiesResponse, ControllerServiceCapability, CreateVolumeResponse,
+            DeleteVolumeResponse, Topology, Volume,
+        },
+    },
+    utils::error_full_message,
+};
+use serde::{de::IntoDeserializer, Deserialize};
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::{
+    k8s_openapi::api::core::v1::PersistentVolumeClaim, kube::runtime::reflector::ObjectRef,
+};
+use tonic::{Request, Response, Status};
+use uuid::Uuid;
+
+pub const TOPOLOGY_NODE: &str = "secrets.stackable.tech/node";
+
+#[derive(Snafu, Debug)]
+#[snafu(module)]
+enum CreateVolumeError {
+    #[snafu(display("failed to parse CreateVolume parameters"))]
+    InvalidParams { source: serde::de::value::Error },
+    #[snafu(display("failed to load {pvc}"))]
+    FindPvc {
+        source: stackable_operator::error::Error,
+        pvc: ObjectRef<PersistentVolumeClaim>,
+    },
+    #[snafu(display("failed to resolve owning Pod of {pvc}"))]
+    ResolveOwnerPod {
+        pvc: ObjectRef<PersistentVolumeClaim>,
+    },
+    #[snafu(display("failed to parse secret selector from annotations of {pvc}"))]
+    InvalidSecretSelector {
+        source: serde::de::value::Error,
+        pvc: ObjectRef<PersistentVolumeClaim>,
+    },
+    #[snafu(display("failed to initialize backend"))]
+    InitBackend {
+        source: backend::dynamic::FromSelectorError,
+    },
+    #[snafu(display("failed to find nodes matching scopes"))]
+    FindNodes { source: backend::dynamic::DynError },
+    #[snafu(display("no nodes match scopes"))]
+    NoMatchingNode,
+}
+
+impl From<CreateVolumeError> for Status {
+    fn from(err: CreateVolumeError) -> Self {
+        let full_msg = error_full_message(&err);
+        // Convert to an appropriate tonic::Status representation and include full error message
+        match err {
+            CreateVolumeError::InvalidParams { .. } => Status::invalid_argument(full_msg),
+            CreateVolumeError::FindPvc { .. } => Status::unavailable(full_msg),
```

Comment:
I guess same question here
```suggestion
            CreateVolumeError::FindPvc { .. } => Status::not_found(full_msg),
```